File: Leagacy/threasholding.py
import numpy as np
from PIL import Image

# Load the image
img_path = 'speed_photos/Beeswing.png'
img = Image.open(img_path)
img_array = np.array(img)

# Separate the red channel
red_channel = img_array[:, :, 0]

# Threshold the red channel
threshold_value = 20 # This is an arbitrary threshold, adjust as needed
binary_image = red_channel > threshold_value

# Create a new image where only the red is visible and convert binary mask to uint8
red_visible = np.zeros_like(img_array)
red_visible[:, :, 0] = red_channel * binary_image.astype(np.uint8)

# display the image
Image.fromarray(red_visible).show()

File: Leagacy/threasholding.m
clear all;
close all;

% Read the image
img = imread('speed_photos/Beeswing.png');

% Separate the red channel
red_channel = img(:, :, 1);

% Threshold the red channel
threshold_value = 120; % This is an arbitrary threshold, adjust as needed
binary_image = red_channel > threshold_value;

% Convert binary_image to type double for multiplication
binary_image = double(binary_image);

% Create a new image where only the red is visible
% The result is a 3D matrix where the first plane is the red channel,
% and the other two are black.
red_visible = cat(3, red_channel .* binary_image, zeros(size(binary_image)), zeros(size(binary_image)));

% Display the image
imshow(red_visible);
